### ISSUE TITLE
Template var refereneces are not marked as eslintUsed

### ISSUE THREAD

**Issue by Trinovantes** [vuejs/vue-eslint-parser#287]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [x] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v10.0.0

### What version of `eslint-plugin-vue` and `vue-eslint-parser` are you using?

- vue-eslint-parser@10.7.0
- eslint-plugin-vue@10.7.0

### What did you do?

eslint 10 recommended settings now enables `no-useless-assignment` rule which triggers false positives in SFCs

From what I understand in this thread https://github.com/eslint/eslint/issues/20501, I think template variables need to be marked as `eslintUsed`

### What did you expect to happen?

No errors

### What actually happened?

```
/home/stephen/eslintjs10vuebug/src/MyComponent.vue
  4:7  error  This assigned value is not used in subsequent statements  no-useless-assignment
```

### Link to Minimal Reproducible Example

https://github.com/Trinovantes/eslintjs10vuebug
